fix(formatter): missing parenthesis for `TSAsExpression`

Dunqing opened the pull request:
Port from https://github.com/biomejs/biome/blob/4a5ef84930344ae54f3877da36888a954711f4a6/crates/biome_js_syntax/src/parentheses/expression.rs#L702-L709

nnnnoel commented on crates/oxc_formatter/src/parentheses/expression.rs:
```diff
@@ -942,16 +942,23 @@ fn await_or_yield_needs_parens(span: Span, node: &AstNodes<'_>) -> bool {
+    parent: &AstNodes<'_>,
+) -> bool {
+    match parent {
+        AstNodes::ConditionalExpression(_)
+        // Binary-like
+        | AstNodes::LogicalExpression(_)
+        | AstNodes::BinaryExpression(_) => true,
```
Oh i missed, `type_cast_like_needs_parens` already implements `UpdateExpression` case.

Dunqing replied:
Already added in `type_cast_like_needs_parens`, see https://github.com/oxc-project/oxc/blob/960bab4cca1f2e5e30ef6b5f411e73e0d5f1d702/crates/oxc_formatter/src/parentheses/expression.rs#L674-L675

leaysgur reviewed (approved):
👍🏻

Dunqing committed:
fix(formatter): missing parenthesis for `TSAsExpression` (#13842)

Port from https://github.com/biomejs/biome/blob/4a5ef84930344ae54f3877da36888a954711f4a6/crates/biome_js_syntax/src/parentheses/expression.rs#L702-L709
